### PR TITLE
feat: Added hints for function arguments

### LINKED ISSUE #4789: [Feature]: JS function arguments tooltip
Objective: Show function arguments in the autocomplete suggestions and allow users to configure parameters upon selecting an autocomplete result. 

[LOOM DEMO](https://www.loom.com/share/5869e7a7c8d34a91b5a93b61c5c98baa)

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
Adds a tooltip that shows the function signatures when the cursor is between ( ) to help complete function arguments.
https://www.figma.com/file/5QitVVvqgEc6nhR7SbdinY/ADS2.0---Tokens?type=design&node-id=7639-124926&mode=dev

#### PR fixes following issue(s)
Fixes #4789
>
#### Media
Screenshot 2023-11-06 at 12 10 42

#### Type of change
- New feature (non-breaking change which adds functionality)
>
>
#### How Has This Been Tested?
- [x] Manual
- [x] Cypress
>
>
#### Test Plan
- [x] Function signature tooltip should show when the cursor is in between `( )`.
- [x] Tooltip should disappear when the cursor is moved outside '( )`.
- [x] Tooltip should disappear when peek over popover is triggered.
- [x] Tooltip appear as soon as a function is selected from autocomplete results.
- [x] Evaluated Value popover should disappear when tooltip popover is open.

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/utils/autocomplete/CodemirrorTernService.ts [hunk 1]
```diff
@@ -198,8 +212,167 @@ class CodeMirrorTernService {
     });
   }
 
+  updateArgHints(cm: CodeMirror.Editor) {
```
Comment (arunvjn) [inline app/client/src/utils/autocomplete/CodemirrorTernService.ts:215]: Copied over from codemirror-tern plugin

File: app/client/src/globalStyles/CodemirrorHintStyles.ts [hunk 2]
```diff
@@ -122,6 +122,7 @@ export const CodemirrorHintStyles = createGlobalStyle<{
 
     .CodeMirror-hint-header {
       padding: 8px;
+      font-size: 11px;
```
Comment (ohansFavour) [inline app/client/src/globalStyles/CodemirrorHintStyles.ts:125]: Can you consider ADS variables for font-sizes?.

Example: https://github.com/appsmithorg/appsmith/blob/eaac70d9f7c6ce39caf123e9a3db1a425d6f907d/app/client/src/components/editorComponents/RadioButtonGroup.tsx#L51

File: app/client/src/globalStyles/CodemirrorHintStyles.ts [hunk 3]
```diff
@@ -156,6 +157,7 @@ export const CodemirrorHintStyles = createGlobalStyle<{
     .CodeMirror-Tern-completion {
       font-family: ${(props) => props.theme.fonts.code};
       display: flex;
+      font-size: 12px;
```
Comment (ohansFavour) [inline app/client/src/globalStyles/CodemirrorHintStyles.ts:160]: Same as above

File: app/client/src/globalStyles/CodemirrorHintStyles.ts [hunk 4]
```diff
@@ -254,6 +253,33 @@ export const CodemirrorHintStyles = createGlobalStyle<{
     }
   }
 
+  .CodeMirror-Tern-tooltip {
+    font-family: ${(props) => props.theme.fonts.code};
+    &&& {
+      font-size: 12px;
```
Comment (ohansFavour) [inline app/client/src/globalStyles/CodemirrorHintStyles.ts:259]: Same as above

File: app/client/src/utils/autocomplete/CodemirrorTernService.ts (hunk 1 above)
Comment (ohansFavour) [reply app/client/src/utils/autocomplete/CodemirrorTernService.ts:215]: Cool, can you also add this comment to the code (and maybe a link to where it was copied from)?. That's our usual practice.

File: app/client/src/utils/autocomplete/CodemirrorTernService.ts [hunk 5]
```diff
@@ -239,9 +412,7 @@ class CodeMirrorTernService {
     resolve: any,
   ) {
     if (error) return this.showError(cm, error);
-    if (data.completions.length === 0) {
-      return this.showError(cm, "No suggestions");
-    }
+    if (data.completions.length === 0) return;
```
Comment (ohansFavour) [inline app/client/src/utils/autocomplete/CodemirrorTernService.ts:415]: This makes sense!

**Review by ohansFavour — APPROVED** [review]
The changes look good. Minor suggestions added.

**Review by ohansFavour — DISMISSED** [review]

**Pull request merged**